Gcn

Chahalprincy opened the pull request:
## Description

Implement usage of `DCLightningModule` with `GCNModel`

Fix #(issue)

<!-- Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.
List any dependencies that are required for this change. -->


## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

rbharath commented on deepchem/models/lightning/dc_lightning_module.py:
```diff
@@ -0,0 +1,99 @@
+class DCLightningModule(pl.LightningModule):
+  """DeepChem Lightning Module to be used with Lightning trainer.
+
+  Example code
```
Make sure to match numpy doc style exactly or the code won't render in the docs

https://numpydoc.readthedocs.io/en/latest/format.html

rbharath commented on deepchem/models/lightning/dc_lightning_module.py:
```diff
@@ -0,0 +1,99 @@
+  PYTORCH_LIGHTNING_IMPORT_FAILED = True
+
+
+class DCLightningModule(pl.LightningModule):
```
Can you add this to the `docs/` folder so docs are generated for this new class?

rbharath commented on deepchem/models/lightning/dc_lightning_module.py:
```diff
@@ -0,0 +1,99 @@
+  >>> from deepchem.models.lightning.dc_lightning_module
+  ...   import DCLightningModule
+  >>> model = MultitaskClassifier(params)
+  >>> valid_dataloader = DataLoader(test_dataset)
```
`test_dataset` isn't defined? I think this might break on doctest. Try running doctest locally with

```
python -m doctest dc_lightning_module.py
```

rbharath commented on deepchem/models/lightning/dc_lightning_module.py:
```diff
@@ -0,0 +1,99 @@
+
+  def configure_optimizers(self):
+    """Configure optimizers, for details refer to:
+    https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.core.LightningModule.html?highlight=LightningModule
```
You should probably add the link to a References section in the main docstring to render it nicely

rbharath commented on deepchem/models/lightning/dc_lightning_module.py:
```diff
@@ -0,0 +1,99 @@
+    """For details refer to:
+    https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.core.LightningModule.html?highlight=LightningModule  # noqa
+
+    Args:
```
Needs to be converted to numpy doc style

rbharath reviewed (commented):
Sorry for the multiple single comments. I thought I was just going to make one comment but ended up reviewing.

The code should use `make_pytorch_dataset` instead of requiring custom `Dataset` classes to handle DeepChem datasets. Once that's fixed, and doc style is fixed I think this PR will be good. Nice work! Looking forward to seeing this come together

Chahalprincy commented on deepchem/models/lightning/tests/test_dc_lightning_module.py:
```diff
@@ -0,0 +1,120 @@
+  PYTORCH_LIGHTNING_IMPORT_FAILED = True
+
+
+class TestMultiTaskDataset(torch.utils.data.Dataset):
```
This will resolve once I implement the lightning dataset module this week.

Chahalprincy commented:
Hi, @rbharath , @abhiskk I have addressed all the review comments and none of the tests now are failing because of lightning. Please take a look when you get a chance.

rbharath commented on deepchem/models/lightning/dc_lightning_module.py:
```diff
@@ -44,36 +37,32 @@ def __init__(self, dc_model):
+
+    Parameters
+    ----------
+    batch (:class:`~torch.Tensor` | (:class:`~torch.Tensor`, ...) | [:class:`~torch.Tensor`, ...]):
```
I don't think this is numpydoc style. Could you please verify and fix?

rbharath commented on deepchem/models/lightning/tests/test_dc_lightning_module.py:
```diff
@@ -56,10 +56,61 @@ def __getitem__(self, index):
+                   'PyTorch Lightning is not installed')
+  def test_gcn_model(self):
+
+    class TestGCNDataset(torch.utils.data.Dataset):
```
You don't need this class. Instead you can just use `make_pytorch_dataset()` called directly from the deepchem dataset

abhiskk replied:
hey @rbharath, one comment on this, Princy is working on a dataset module which will make this implementation easy to use with the `make_pytorch_dataset()` functionality from deepchem, my suggestion to Princy was to have this dataset as it is right now because she has setup the interaction with lightning module in a particular way, this will definitely be made simpler in another PR later this week.

rbharath replied:
Sounds good!
